cyberark/secretless-broker
# Unify and standardize plugin API interaction points

Partial impl of #64 

[Jenkins Build](https://jenkins.conjur.net/view/conjurinc/job/conjurinc--secretless/job/64-handlers-as-plugins-v2/)

What this PR does:
- Makes Manager interface a first-class citizen
- Exports correct interfaces of handlers and interfaces in a plugin
- Unifies versioned interfaces into a versioned plugin API version module
- Ensures we use our API module definition when referring to interfaces internally
- Bump API version to 0.0.2
- Adds ability to store listeners and handler in plugin manager
  - Note: Does not have them used yet

How to test:
- In server dev env with with matching server/secretless code:
  - Run build_plugin
  - Run build_server
  - Run build_client
  - Run `secretless-server` in one terminal
  - Run `secretless` in another terminal and ensure that the plugin is visible in server UI
